crates/fluvio-stream-dispatcher/src/metadata/local.rs

```diff
@@ -309,15 +308,17 @@ cfg_if::cfg_if! {
                 Self { path, stores }
             }
 
-            async fn get_store<S: Spec + DeserializeOwned>(&self) -> Result<Arc<SpecStore>> {
+            fn get_store<S: Spec + DeserializeOwned>(&self) -> Result<Arc<SpecStore>> {
                 let key = S::LABEL;
-                let read = self.stores.upgradable_read().await;
+                let read = self.stores.read();
                 Ok(match read.get(key) {
                     Some(store) => store.clone(),
                     None => {
-                        let mut write = RwLockUpgradableReadGuard::upgrade(read).await;
-                        let store = Arc::new(SpecStore::load::<S, _>(self.path.join(key)).await?);
+                        drop(read);
+                        let mut write = self.stores.write();
+                        let store = Arc::new(SpecStore::load::<S, _>(self.path.join(key))?);
```

Comment:
we need to move this out of here but it's not blocker.  let's create issue for it

Reply:
Good catch! I forgot to remove it

Reply:
https://github.com/infinyon/fluvio/issues/4177